src/Core/Domain/Order/Invoice/Command/UpdateInvoiceNoteCommand.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Order\Invoice\Command;
```

Comment:
@matks this is something I want to discuss with you. 

I've added `Invoice` subdomain as child for `Domain/Order` instead of putting it at the root (e.g. `Domain/Invoice`). At the moment it has only one `UpdateInvoiceNoteCommand` command, but I could move `GenerateOrderInvoiceCommand` (and rename to simply `GenerateInvoiceCommand`) under this subdomain as well. 

This kind of adds up for me, because Invoicing does not make sense without Orders, however, I can see that Invoicing could potentially live on it's own as well.

So, wdyt? Should I keep `Domain/Order/Invoice` or move it to `Domain/Invoice`? :thinking:

Reply:
OK, this one is a tough one.

To answer it, I went to check the Orders > Invoices BO page. From what I see, I think that *for PrestaShop*, invoices without orders mean nothing.

There's no accounting lifecycle, there is no invoices payment cycle or management, I dont see any complex invoices actions.

Invoices seem to be only an order subdomain for now, so let's for for `Domain/Order/Invoice`

Reply:
Ok, this makes sense for me too. :+1: 

However, there's more to it. Currently there a quite a few commands in `Domain/Order/Command` (there are more in other PR as well) that are not directly interacting with `Order`, but rather with order `Product` aka `OrderDetail` aka `OrderLine`. 

So I think that we could create another subdomain for `Order` which in my opinion makes sense since `Order` is quite a bit module/feature/whatever. It could be called `Domain/Order/OrderDetail` if we want to follow current naming or `Domain/Order/OrderLine` which is also a common name for it in ecommerce.

wdyt @matks?

Reply:
Not sure about this one 🤔 but I guess if we dont do it, we're going to have a huuuuuge `Order` domain, right ?

Reply:
Yeah, it's very likely.

Reply:
Then it's a yes 👍